omod/src/test/java/org/openmrs/module/metadatamapping/web/rest/MetadataTermMappingResourceOperationTest.java

```diff
@@ -0,0 +1,277 @@
+		    "3cfcf118-931c-46f7-8ff6-7b876f0d4202"));
+		
+		// when
+		SimpleObject postResponseData = deserialize(handle(newPostRequest(getURI(), postData)));
+		
+		// then
+		assertNotNull(postResponseData);
+		Object newUuid = PropertyUtils.getProperty(postResponseData, "uuid");
+		assertNotNull(newUuid);
+		
+		SimpleObject getResponseData = deserialize(handle(newGetRequest(getURI() + "/" + newUuid)));
+		assertNotNull(getResponseData);
+		assertEquals("term-123", PropertyUtils.getProperty(getResponseData, "code"));
+		assertEquals("Test Term Mapping 123", PropertyUtils.getProperty(getResponseData, "name"));
+		Object metadataSource = getResponseData.get("metadataSource");
+		assertEquals("df29a160-0add-4598-8ac2-b11a9eb3cdb8", PropertyUtils.getProperty(metadataSource, "uuid"));
+		
+		Object mappedObject = getResponseData.get("mappedObject");
+		assertEquals("org.openmrs.Drug", PropertyUtils.getProperty(mappedObject, "className"));
+		assertEquals("3cfcf118-931c-46f7-8ff6-7b876f0d4202", PropertyUtils.getProperty(mappedObject, "uuid"));
+	}
+	
+	@Test
+	public void update_shouldPersist() throws Exception {
+		// given
+		SimpleObject postData = new SimpleObject().add("description", "This is the new term mapping description").add(
+		    "mappedObject",
+		    new SimpleObject().add("className", "org.openmrs.Drug").add("uuid", "3cfcf118-931c-46f7-8ff6-7b876f0d4202"));
+		
+		// when
+		MockHttpServletResponse postResponse = handle(newPostRequest(getURI() + "/" + getUuid(), postData));
+		
+		// then
+		assertNotNull(postResponse);
+		
+		SimpleObject getResponseData = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertNotNull(getResponseData);
+		assertEquals("This is the new term mapping description", PropertyUtils.getProperty(getResponseData, "description"));
+		
+		Object mappedObject = getResponseData.get("mappedObject");
+		assertEquals("org.openmrs.Drug", PropertyUtils.getProperty(mappedObject, "className"));
+		assertEquals("3cfcf118-931c-46f7-8ff6-7b876f0d4202", PropertyUtils.getProperty(mappedObject, "uuid"));
+	}
+	
+	@Test
+	public void delete_shouldRetire() throws Exception {
+		// given
+		SimpleObject unRetiredTermMapping = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertEquals(Boolean.FALSE, PropertyUtils.getProperty(unRetiredTermMapping, "retired"));
+		
+		// when
+		MockHttpServletResponse deleteResponse = handle(newDeleteRequest(getURI() + "/" + getUuid()));
+		
+		// then
+		assertNotNull(deleteResponse);
+		
+		SimpleObject retiredTermMapping = deserialize(handle(newGetRequest(getURI() + "/" + getUuid())));
+		assertEquals(Boolean.TRUE, PropertyUtils.getProperty(retiredTermMapping, "retired"));
+	}
+	
+	@Test
+	public void getSingle_shouldGetWithCodeAndSourceUuid() throws Exception {
+		// given
+		// test data
+		
+		// when
+		MockHttpServletRequest request = newGetRequest(getURI() + "/drug-tri");
```

Comment:
This should really be /metadatatermmapping?code=drug-tri&source=9cace0bd-6f2a-4cc3-a26d-6fa292f1f2c1

I think we could actually make it just /term since the full resource url looks like /ws/rest/v1/metadatamapping/term?